pyne/vta.py

```diff
@@ -0,0 +1,551 @@
+    d_z = end.z - start.z
+    dist = math.sqrt(d_x * d_x + d_y * d_y + d_z * d_z)
+    v_x = d_x / dist
+    v_y = d_y / dist
+    v_z = d_z / dist
+    return [v_x, v_y, v_z]
+
+def _find_next_grid_1d(cor, v_1d, bound):
+    """
+    Find the next grid coordinate of a direction.
+
+    Parameters:
+    -----------
+    cor : float
+        Coordinate of current point in the direction.
+    v_1d : float
+        Direction value.
+    bound : list of float
+        Boundary values of current direction.
+
+    Return:
+    -------
+    n_grid : float
+        The coordinate value of next mesh grid.
+    """
+    n_grid = 0.0
+    flag = False
+    if v_1d > 0:
+        for i in range(len(bound)):
+            if cor < bound[i]:
+                n_grid = bound[i]
+                flag = True
+                break
+    else:
+        for i in range(len(bound)-1, -1, -1):
+            if cor > bound[i]:
+                n_grid = bound[i]
+                flag = True
+                break
+    if flag:
+        return n_grid
+    else:
+        # point goes out of the mesh
+        return v_1d * float('inf')
+
+
+def _cal_max_travel_length_in_current_voxel(tp, end, v, bounds):
+    """
+    Calculate the maximum travel length in current voxel before reach next
+    boundary.
+    
+    Parameters:
+    -----------
+    tp : Point
+        Temporary point position.
+    end : Point
+        End point
+    v : list
+        Direction vector from start to end: [v_x, v_y, v_z].
+    bounds : list
+        Boundaries of the mesh grid.
+
+    Return:
+    -------
+    t_maxs : list
+        The maximum travel lenght in current voxel for 3 directions. Format: 
+        [t_max_x, t_max_y, t_max_z]. These value could be both positive and
+        negtive. And could be 'inf' or '-inf'.
+    """
+    n_x_grid = _find_next_grid_1d(tp.x, v[0], bounds[0])
+    n_y_grid = _find_next_grid_1d(tp.y, v[1], bounds[1])
+    if v[0] == 0.0:
+        t_max_x = float('inf')
+    else:
+        t_max_x = (n_x_grid - tp.x) / v[0]
+    if v[1] == 0.0:
+        t_max_y = float('inf')
+    else:
+        t_max_y = (n_y_grid - tp.y) / v[1]
+    return [t_max_x, t_max_y]
+
+def _move_to_next_voxel(x_idx, y_idx, z_idx, tp, t_temp, t_maxs, v):
+    """
+    Move the point to next voxel.
+
+    Parameters:
+    -----------
+    x_idx : int
+        Mesh idx in x direction.
+    y_idx : int
+        Mesh idx in y direction.
+    z_idx : int
+        Mesh idx in z direction.
+    tp : Point
+        Temporary point position.
+    t_temp: float
+        Temporary travel lenght from the start.
+    t_maxs: list
+        The maximum travel length in current voxel for 3 directions.
+    v : list
+        Direction vector from start to end.
+
+    Returns:
+    --------
+    x_idx : int
+        Modified x_idx.
+    y_idx : int
+        Modified y_idx.
+    z_idx : int
+        Modified z_idx.
+    tp : Point
+        Point after moving along the direction of v.
+    t_temp : float
```

Comment:
Does this mean an updated length of the ray?

Reply:
It might be nice to provide some extra clarity about this. If I understand correctly, the returned `t_temp` value is the total distance traveled from the start of the ray, but that isn't used or provided as part of this function, which is a bit confusing.

I'll suggest something like "Updated travel length since the beginning of the ray." I'm not sure if that's better really 😞

Also, having different names for outgoing variables might be useful to provide clarity here.